rfcs/0165-Anonymous-scopes.md

```diff
@@ -0,0 +1,48 @@
+# RFC 165 - Anonymous scopes
+* Comments: [#165](https://github.com/taskcluster/taskcluster-rfcs/pull/165)
+* Proposed by: @ricky26
+
+# Summary
+
+We currently assign no scopes to calls with no credentials. With this proposal
+we would assign a single role to all calls and any scopes expanded from that.
+Additionally, we will move all API calls behind at least one scope.
+
+## Motivation
+
+This is primarily aimed at private TaskCluster installations. With this, it will
+be possible to prevent any useful access to the TaskCluster installation without
+credentials.
+
+# Details
+## Implementation
+A single scope will be assumed for all calls (with or without credentials):
+`assume:anonymous`. A new API call will be added to the auth service
+(`authenticateAnonymous`) which will return `{"scopes":...,"expiry":...}`.
+The authentication middleware will call `authenticateAnonymous` but cache the
```

Comment:
```suggestion
The authentication middleware in each service will call `authenticateAnonymous` but cache the
```
(just reflecting my understanding from the conversation)

Reply:
Yes, absolutely! I've just committed that suggestion as it is what I was aiming for.